### PR TITLE
Add ESLint + Prettier for improved dev workflow

### PR DESCRIPTION
**Pull request by ItsJonQ**
## Add ESLint + Prettier for improved dev workflow

This update adds [ESLint](https://eslint.org/) and [Prettier](https://prettier.io/) to the development workflow of Anime.js 🎉 !

These updates will help keep the code consistent and make it easier for others to contribute. Bonus: Those with editors like Visual Code (with the proper setup) can auto-format on save with Prettier 🙌 

### Inspired by Parcel

The configs for both ESLint and Prettier very light touch, and were inspired by the ones defined by [Parcel](https://github.com/parcel-bundler/parcel), a very popular zero-config bundler. The reason for this decision was to avoid complexity and bloat. I did, however, make some modifications:

### ESLint

Added globally defined variables like `NodeList`, `window`, and others available in a standard non-Node based environment. I've also added `sourceType: "module"` since the original `src/index.js` uses `import/export` instead of `require/module`.

### Lint Staged

Again, inspired by Parcel, I've added a 3rd library [lint-staged](https://www.npmjs.com/package/lint-staged) to automatically format any `src/` `.js` file with prettier when staging commits to Git. Very handy :)

### Formatted

I linted + formatted the main `src/index.js`. Since Prettier fixed up the entire file, the diffs are (unfortunately) difficult to make out. There were only a couple of lint issues, which included removing unused variables and unnecessary escapes in the regex.

### Omitted Changes

To respect the current project setup, I did NOT include my local generated `package-lock.json`. I also did **not** rebuild the `lib/` files. Although, I did test this, and the bundling worked as expected.

### Testing

The ESLint + Prettier formatting should not affect Rollup's bundling. To double check, I ran `npm run build` and verified the `document/index.html` examples in my browsers. From what I can see, the animations work as expected, without errors in console.

@juliangarnier Hope this helps!!! I decided to make this PR against master since these updates aren't related to memory leaks. I can modify this if you prefer.

Issue reference:
https://github.com/juliangarnier/anime/issues/463

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "extends": "eslint:recommended",
+  "parserOptions": {
+    "ecmaVersion": 8,
+    "sourceType": "module"
+  },
+  "env": {
+    "node": true,
```
Comment (mubaidr) [inline .eslintrc.json]: Set this to browser, we are writing script for browser only.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    "es6": true
+  },
+  "globals": {
```
Comment (mubaidr) [inline .eslintrc.json]: No need to specify these globals, these will be automatically handled by eslint when `env` is set to `browser`

File: .prettierrc [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
```
Comment (mubaidr) [inline .prettierrc]: Should be `false` to follow current coding practices

File: .eslintrc.json (hunk 1 above)
Comment (ItsJonQ) [reply .eslintrc.json]: Oo! Thanks for this + tips on `globals`!

File: .prettierrc (hunk 3 above)
Comment (ItsJonQ) [reply .prettierrc]: @mubaidr I had original set this to `false`. When I checked the diffs, I noticed there was a mixture of single and double quotes, like this:

https://github.com/juliangarnier/anime/blob/master/src/index.js#L26

@juliangarnier We can go with either one! Just let us know :)

File: .eslintrc.json (hunk 1 above)
Comment (ItsJonQ) [reply .eslintrc.json]: @mubaidr Made the updates :D 

https://github.com/juliangarnier/anime/pull/479/commits/8b9f6974fc9fbc4b9fe23394f312110afe0b00c5

🙌

File: .prettierrc (hunk 3 above)
Comment (mubaidr) [reply .prettierrc]: In this case, I vote for `'singleQuote': true` 👍

**Pull request merged**